packages/quantic/cypress/e2e/default-1/case-classification/case-classification.cypress.ts

```diff
@@ -889,4 +893,343 @@ describe('quantic-case-classification', () => {
+          Expect.correctSugestionsOrder(allOptions.slice(0, suggestionsCount));
+          Expect.numberOfInlineOptions(0);
+          Expect.logClickedSuggestion(firstSuggestionIndex, true);
+          Expect.correctValue(allOptions[firstSuggestionIndex].value);
+        });
+
+        scope('when selecting a suggestion', () => {
+          const clickedIndex = 1;
+
+          Actions.clickSuggestion(clickedIndex);
+          Expect.logClickedSuggestion(clickedIndex);
+          Expect.logUpdatedClassificationFromSuggestion(
+            coveoDefaultField,
+            clickedIndex
+          );
+          Expect.correctValue(allOptions[clickedIndex].value);
+        });
+
+        scope('when selecting an option from the select input', () => {
+          const clickedIndex = 3;
+          Actions.clickSelectTitle();
+          Actions.openSelectInput();
+          Actions.clickSelectOption(clickedIndex);
+          Expect.correctValue(allOptions[clickedIndex].value);
+          Expect.logUpdatedClassificationFromSelectOption(
+            coveoDefaultField,
+            clickedIndex
+          );
+          Expect.fetchClassificationsAfterValueChange(
+            coveoDefaultField,
+            allOptions.slice(0, suggestionsCount)
+          );
+        });
+      });
+    });
+  });
+
+  describe('when a field value changes and the fetchDocumentSuggestionOnChange property is set to true', async () => {
+    const mockDocuments = {
+      documents: [
+        {
+          clickUri: 'https://example.com',
+          excerpts: 'Roads are just a suggestion Marge, just like pants.',
+          fields: {},
+          hasHtmlVersion: true,
+          title: 'homer philosophy vol 1',
+          uniqueId: '001',
+        },
+        {
+          clickUri: 'https://example.com',
+          excerpts:
+            'If God didnt want me to eat chicken in church, then he would have made gluttony a sin.',
+          fields: {},
+          hasHtmlVersion: true,
+          title: 'homer philosophy vol 2',
+          uniqueId: '002',
+        },
+      ],
+      responseId: '123',
+      totalCount: 2,
+    };
+
+    describe('when a suggestion is selected', () => {
+      it('should automatically fetch new document suggestions', () => {
+        const clickedIndex = 1;
+        const suggestionsCount = 3;
+        visitCaseClassification({
+          fetchDocumentSuggestionOnChange: true,
+        });
+        scope('when loading the page', () => {
+          Expect.displayLabel(true);
+          Expect.numberOfInlineOptions(0);
+          Expect.numberOfSuggestions(0);
+          Expect.displaySelectTitle(false);
+          Expect.displaySelectInput(true);
+        });
+
+        scope('when fetching suggestions', () => {
+          const firstSuggestionIndex = 0;
+
+          mockCaseClassification(
+            coveoDefaultField,
+            allOptions.slice(0, suggestionsCount)
+          );
+          mockDocumentSuggestion(mockDocuments.documents);
```

Comment:
Why do we need to mock the document suggestions returned?
The only thing that we need to test is that there is a document suggestion request made  whenever a case classification is selected.
The content of the document suggestion doesn't really matter thus I don't think we need to mock the document suggestions.

Reply:
yeah if we dont test for response doesnt make sense then to mock documents